test/powershell/Modules/Microsoft.PowerShell.Management/Get-Location.Tests.ps1

```diff
@@ -2,20 +2,80 @@
 # Licensed under the MIT License.
 Describe "Get-Location" -Tags "CI" {
     $currentDirectory=[System.IO.Directory]::GetCurrentDirectory()
+
     BeforeEach {
-	Push-Location $currentDirectory
+        Push-Location $currentDirectory
     }
 
     AfterEach {
-	Pop-location
+	    Pop-location
     }
 
     It "Should list the output of the current working directory" {
-
-	(Get-Location).Path | Should -BeExactly $currentDirectory
+	    (Get-Location).Path | Should -BeExactly $currentDirectory
     }
 
     It "Should do exactly the same thing as its alias" {
-	(pwd).Path | Should -BeExactly (Get-Location).Path
+	    (pwd).Path | Should -BeExactly (Get-Location).Path
+    }
+
+    It "Should throw an exception when missing an argument for parameter 'PSProvider'" {
+         { Get-Location -PSProvider } | Should -Throw -ErrorId "MissingArgument"
+    }
+
+    It "Should throw an exception when missing an argument for parameter 'PSDrive'" {
+        { Get-Location -PSDrive } | Should -Throw -ErrorId "MissingArgument"
+    }
+
+    It "Should throw an exception when missing an argument for parameter 'StackName'" {
+        { Get-Location -StackName } | Should -Throw -ErrorId "MissingArgument"
+    }
+
+    It "Should return a PathInfo object when no parameters are given" {
+        Get-Location | Should -BeOfType System.Management.Automation.PathInfo
+    }
+
+    It "Should return a PathInfo object given a valid argument for parameter 'PSDrive'" {
+        $tempPSDriveName = "GeLocationTempPSDrive"
+
+        $tempPSDrive = New-PSDrive -Name $tempPSDriveName -PSProvider "FileSystem" -Root $currentDirectory
+
+        Get-Location -PSDrive $tempPSDriveName | Should -BeOfType System.Management.Automation.PathInfo
+
+        $tempPSDrive | Remove-PSDrive
```

Comment:
If we should cleanup it is better to use try-filnally and put `$tempPSDrive | Remove-PSDrive` to finally block.